### PR TITLE
Move documentation to docs folder

### PR DESCRIPTION
**Pull request by benbalter**
Per https://github.com/jekyll/jekyll/issues/5409, this pull requests moves the user-facing documentation from the `site` folder to the `docs` folder where we can rely on Pages to automatically generate the docs for us.

To do this, I did a few things:
1. Abstract the docs folder to a variable in the rake task
2. Move the rake task to use the docs folder rather than the site folder
3. Moved the actual folder content
4. Migrated the maintainer docs, which was at `/docs` to `/docs/_docs/development/`, and added YAML front matter. 

There should be no other change to the user-facing site content.

Also to note, `rake site:publish` simply runs `rake site:generated_pages` and commits and pushes the result (since we no longer need to do the multiple branch dance). It may be possible to have Jekyllbot do this after each release (to update the history and contributing files).

Once this merges, we can delete the `gh-pages` branch. We're also working to get this repo early access to new releases of Jekyll, so that it can always be running the latest stable version, regardless of what version other Pages users are using.

Fixes https://github.com/jekyll/jekyll/issues/5409.

### CODE REVIEW

**Comment by zdroid** [comment]
> We're also working to get this repo early access to new releases of Jekyll, so that it can always be running the latest stable version, regardless of what version other Pages users are using.

Exactly why this change is awesome. Big :+1:.

File: rake/site.rake [hunk 1]
```diff
@@ -37,67 +37,29 @@ namespace :site do
...
     Dir.chdir('gh-pages') do
-      sh "git add ."
-      sh "git commit --allow-empty -m 'Updating to #{sha}.'"
-      sh "git push origin gh-pages"
+      sh "git add docs/"
+      sh "git commit --allow-empty -m 'Generating pages for #{sha}.'"
+      sh "git push origin master"
     end
     puts 'Done.'
   end
```
Comment (parkr) [inline rake/site.rake]: This can be removed. I'll just change the settings so it uses `master /docs`.

File: rake/site.rake [hunk 2]
```diff
@@ -37,67 +37,29 @@ namespace :site do
...
-      "destination"  => File.expand_path("gh-pages"),
-      "sass"         => { "style" => "compressed" }
-    })
-
-    File.open('gh-pages/.nojekyll', 'wb') { |f| f.puts(":dog: food.") }
-
-    # Commit and push.
     puts "Committing and pushing to GitHub Pages..."
     sha = `git rev-parse HEAD`.strip
     Dir.chdir('gh-pages') do
```
Comment (parkr) [inline rake/site.rake]: Or get rid of this.

File: docs/_data/docs.yml [hunk 3]
```diff
@@ -46,5 +46,6 @@
 - title: Meta
   docs:
   - contributing
+  - development
```
Comment (parkr) [inline docs/_data/docs.yml]: I think I'd like this to be called `maintaining` or something like that. @jekyll/affinity-team-captains, how do you feel about putting your docs on the jekyllrb.com website?

**Comment by ashmaroli** [comment]
I have a personal request:
Can we consider renaming all existing `.markdown` posts to `.md` before implementing this PR?
I've a PR at #5176 dealing with this..
Thanks

**Comment by zdroid** [comment]
I've got some time to review all changes, so I did it now. The changes seem fine to me. Just remove the last `gh-pages` task, as @parkr has noted.

File: docs/_docs/development/index.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
+layout: docs
+title: Maintaining Jekyll
+permalink: /docs/development/
```
Comment (DirtyF) [inline docs/_docs/development/index.md]: Permalinks of this section should all start with `/docs/maintaining/` if we go with @parkr proposal

**Comment by zdroid** [comment]
@benbalter Any progress on this? The number of conflicted files is increasing, so it'll be easier to finish and merge it now than later.

**Comment by DirtyF** [comment]
@zdroid Subscribe to the issue if you want to follow progress, but please don't put pressure on maintainers, we're all volunteers, there is no rush for this.

**Comment by zdroid** [comment]
@DirtyF I did subscribe, don't worry. I'm sorry if I rushed anyone, I know it's hard to keep up with working on open source project. :smile:

**Comment by parkr** [comment]
@zdroid Thanks for the nudge! This is actually waiting on availability of Jekyll 3.3 on GitHub Pages.

**Comment by parkr** [comment]
GitHub Pages is now running Jekyll 3.3. 🎉 https://github.com/blog/2277-what-s-new-in-github-pages-with-jekyll-3-3

File: docs/_config.yml [hunk 5]
```diff
@@ -14,6 +13,9 @@ timezone: America/Los_Angeles
 collections:
   docs:
     output: true
+  posts:
+    permalink: /news/:year/:month/:day/:title/
+    output: true
 
 name: Jekyll • Simple, blog-aware, static sites
```
Comment (zdroid) [inline docs/_config.yml]: Shouldn't `•` be escaped as `&middot;`?

Comment (parkr) [inline docs/_config.yml]: Do you know if it's causing an issue at all? I kind of like the UTF-8 dot. 😄

Comment (XhmikosR) [inline docs/_config.yml]: It should be OK since the HTML files are using utf-8.

**Comment by parkr** [comment]
@jekyllbot: merge +site

**Pull request merged**